modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -195,8 +195,15 @@ final void updateLayerBounds() {
         try {
             crs = CRS.decode(srsName);
         } catch (Exception e) {
-            if (LOGGER.isLoggable(Level.INFO))
-                LOGGER.log(Level.INFO, "Bounds unavailable for layer" + layer);
+            LOGGER.log(
+                    Level.INFO,
+                    () ->
+                            "Default srs ("
+                                    + srsName
+                                    + ") for layer ("
+                                    + layer
+                                    + ") couldn't be decoded. "
+                                    + e.getMessage());
```

Comment:
Do not eat stack traces:

```suggestion
            LOGGER.log(
                    Level.INFO,
                    () ->
                            "Default srs ("
                                    + srsName
                                    + ") for layer ("
                                    + layer
                                    + ") couldn't be decoded. "
                                    , e);
```

Reply:
I disagree: the reason why the code could not be decoded can depend on the set of EPSG plugins available and how they interact with their environment, which may vary by the classpath, network and filesystem conditions (to mention the obvious sources). Removing the stack trace means one would have to exactly reproduce the deployment conditions, leading to a debugging nightmare.

Reply:
In most cases I will agree with you, but for gt-wmts this encode / decode has become somewhat turned around. It isn't the SRS specified by the remote server that are decoded in this function. That decoding is handled by this [function](https://github.com/geotools/geotools/blob/c7b6ac7d1ed6cb68be0497d592d1cbb7c4c3ae88/modules/extension/wmts/src/main/java/org/geotools/ows/wmts/model/TileMatrixSet.java#L66). Instead what is decoded by this function is the result of this [function](https://github.com/geotools/geotools/blob/c7b6ac7d1ed6cb68be0497d592d1cbb7c4c3ae88/modules/extension/wmts/src/main/java/org/geotools/ows/wmts/model/WMTSLayer.java#L105).
Since we're using the CRS.toSRS, I doubt we will have any of those problems you refer to, but sure this is a debugging nightmare.